In [2]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv('C:/Users/cmscomputers/Downloads/Skype/hitters.csv')

In [3]:
df

Name  AtBat  Hits  HmRun  Runs  RBI  Walks  Years  CAtBat  \
0       Andy Allanson    293    66      1    30   29     14      1     293   
1          Alan Ashby    315    81      7    24   38     39     14    3449   
2         Alvin Davis    479   130     18    66   72     76      3    1624   
3        Andre Dawson    496   141     20    65   78     37     11    5628   
4    Andres Galarraga    321    87     10    39   42     30      2     396   
..                ...    ...   ...    ...   ...  ...    ...    ...     ...   
317      Willie McGee    497   127      7    65   48     37      5    2703   
318   Willie Randolph    492   136      5    76   50     94     12    5511   
319    Wayne Tolleson    475   126      3    61   43     52      6    1700   
320     Willie Upshaw    573   144      9    85   60     78      8    3198   
321     Willie Wilson    631   170      9    77   44     31     11    4908   

     CHits  ...  CRuns  CRBI  CWalks  League Division PutOuts  Assists  \
0       66  ...     30    29      14       A        E     446       33   
1      835  ...    321   414     375       N        W     632       43   
2      457  ...    224   266     263       A        W     880       82   
3     1575  ...    828   838     354       N        E     200       11   
4      101  ...     48    46      33       N        E     805       40   
..     ...  ...    ...   ...     ...     ...      ...     ...      ...   
317    806  ...    379   311     138       N        E     325        9   
318   1511  ...    897   451     875       A        E     313      381   
319    433  ...    217    93     146       A        W      37      113   
320    857  ...    470   420     332       A        E    1314      131   
321   1457  ...    775   357     249       A        W     408        4   

     Errors  Salary  NewLeague  
0        20     NaN          A  
1        10   475.0          N  
2        14   480.0          A  
3         3   500.0          N  
4         4    91.5          N  
..      ...     ...        ...  
317       3   700.0          N  
318      20   875.0          A  
319       7   385.0          A  
320      12   960.0          A  
321       3  1000.0          A  

[322 rows x 21 columns]

Scenario 1:
Find of people who have batted for more than 400 matches


In [4]:
(df[(df['AtBat']>400) & (df['Name'])].iloc[:,0:2]).count()

Name     151
AtBat    151
dtype: int64

In [5]:
df[df.AtBat >400].Name.value_counts()

Mitch Webster        1
Bobby Bonilla        1
Spike Owen           1
Jim Presley          1
Darryl Strawberry    1
                    ..
Ken Oberkfell        1
Billy Hatcher        1
Larry Parrish        1
Ray Knight           1
Jody Davis           1
Name: Name, Length: 151, dtype: int64

People batted more than 400 matches are 151.

Scenario 2: 
Find out the name of the player who has hit max home runs

In [6]:
df[df.HmRun==40].iloc[:,:4]

Name  AtBat  Hits  HmRun
136  Jesse Barfield    589   170     40

Scenario 3: 
Find out league wise total number of players


In [7]:
df.groupby(by=['League']).count().iloc[:,0:1]

Name
League      
A        175
N        147

Scenario 6: 
Find out division wise total amount of salary invested

In [8]:
df.groupby(by=['Division']).Salary.sum()

Division
E    80531.006
W    60417.501
Name: Salary, dtype: float64

Scenario 10:
Find out five most senior players.

In [9]:
(df.groupby(by=['Name']).Years.max().iloc[16:]).sort_values(ascending=False).head(5)

Name
Pete Rose         24
Tony Perez        23
Graig Nettles     20
Reggie Jackson    20
Ted Simmons       19
Name: Years, dtype: int64

# 2nd df

In [10]:
gog=pd.read_csv('C:/Users/cmscomputers/Downloads/Skype/Google_Stock_Price_Train.csv')

In [11]:
gog.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1258 entries, 0 to 1257
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    1258 non-null   object 
 1   Open    1258 non-null   float64
 2   High    1258 non-null   float64
 3   Low     1258 non-null   float64
 4   Close   1258 non-null   float64
 5   Volume  1258 non-null   float64
dtypes: float64(5), object(1)
memory usage: 59.1+ KB


Add 4 new columns to this dataframe, containing open, low, high, closing price in Rupees.(right now stock prices are in dollars)

In [13]:
new_gog=(gog.assign(openrs =lambda x: x.Open*75,highrs=lambda x: x.High*75,lowrs=lambda x: x.Low*75,closers=lambda x: x.Close*75))

In [14]:
new_gog

Date    Open    High     Low   Close      Volume    openrs  \
0       1/3/2012  325.25  332.83  324.97  663.59   7380500.0  24393.75   
1       1/4/2012  331.27  333.87  329.08  666.45   5749400.0  24845.25   
2       1/5/2012  329.83  330.75  326.89  657.21   6590300.0  24737.25   
3       1/6/2012  328.34  328.77  323.68  648.24   5405900.0  24625.50   
4       1/9/2012  322.04  322.29  309.46  620.76  11688800.0  24153.00   
...          ...     ...     ...     ...     ...         ...       ...   
1253  12/23/2016  790.90  792.74  787.28  789.91    623400.0  59317.50   
1254  12/27/2016  790.68  797.86  787.66  791.55    789100.0  59301.00   
1255  12/28/2016  793.70  794.23  783.20  785.05   1153800.0  59527.50   
1256  12/29/2016  783.33  785.93  778.92  782.79    744300.0  58749.75   
1257  12/30/2016  782.75  782.78  770.41  771.82   1770000.0  58706.25   

        highrs     lowrs   closers  
0     24962.25  24372.75  49769.25  
1     25040.25  24681.00  49983.75  
2     24806.25  24516.75  49290.75  
3     24657.75  24276.00  48618.00  
4     24171.75  23209.50  46557.00  
...        ...       ...       ...  
1253  59455.50  59046.00  59243.25  
1254  59839.50  59074.50  59366.25  
1255  59567.25  58740.00  58878.75  
1256  58944.75  58419.00  58709.25  
1257  58708.50  57780.75  57886.50  

[1258 rows x 10 columns]

# 3rd df

In [15]:
imd=pd.read_csv('C:/Users/cmscomputers/Downloads/Skype/IMDB-Movie-Data.csv')

In [17]:
imd.head()

Rank                    Title                     Genre  \
0     1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1     2               Prometheus  Adventure,Mystery,Sci-Fi   
2     3                    Split           Horror,Thriller   
3     4                     Sing   Animation,Comedy,Family   
4     5            Suicide Squad  Action,Adventure,Fantasy   

                                         Description              Director  \
0  A group of intergalactic criminals are forced ...            James Gunn   
1  Following clues to the origin of mankind, a te...          Ridley Scott   
2  Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3  In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4  A secret government agency recruits some of th...            David Ayer   

                                              Actors  Year  Runtime (Minutes)  \
0  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014                121   
1  Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012                124   
2  James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016                117   
3  Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016                108   
4  Will Smith, Jared Leto, Margot Robbie, Viola D...  2016                123   

   Rating   Votes  Revenue (Millions)  Metascore  
0     8.1  757074              333.13       76.0  
1     7.0  485820              126.46       65.0  
2     7.3  157606              138.12       62.0  
3     7.2   60545              270.32       59.0  
4     6.2  393727              325.02       40.0

Filter out all action movies

In [ ]:
#imd2=imd.Genre.str.split(',',expand=True)
#imd2.columns=['genre1','genre2','genre3']
#imd2

In [ ]:
#imd=pd.concat([imd,imd2],axis=1)

In [ ]:
imd.info()

In [16]:
(imd[imd['Genre'].str.contains('Action',case=False)])

Rank                           Title                       Genre  \
0       1         Guardians of the Galaxy     Action,Adventure,Sci-Fi   
4       5                   Suicide Squad    Action,Adventure,Fantasy   
5       6                  The Great Wall    Action,Adventure,Fantasy   
8       9              The Lost City of Z  Action,Adventure,Biography   
12     13                       Rogue One     Action,Adventure,Sci-Fi   
..    ...                             ...                         ...   
958   959                  3 Days to Kill       Action,Drama,Thriller   
968   969                         Wrecker      Action,Horror,Thriller   
969   970                 The Lone Ranger    Action,Adventure,Western   
990   991  Underworld: Rise of the Lycans    Action,Adventure,Fantasy   
993   994        Resident Evil: Afterlife     Action,Adventure,Horror   

                                           Description            Director  \
0    A group of intergalactic criminals are forced ...          James Gunn   
4    A secret government agency recruits some of th...          David Ayer   
5    European mercenaries searching for black powde...         Yimou Zhang   
8    A true-life drama, centering on British explor...          James Gray   
12   The Rebel Alliance makes a risky move to steal...      Gareth Edwards   
..                                                 ...                 ...   
958  A dying CIA agent trying to reconnect with his...                 McG   
968  Best friends Emily and Lesley go on a road tri...      Micheal Bafaro   
969  Native American warrior Tonto recounts the unt...      Gore Verbinski   
990  An origins story centered on the centuries-old...  Patrick Tatopoulos   
993  While still out to destroy the evil Umbrella C...  Paul W.S. Anderson   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
5        Matt Damon, Tian Jing, Willem Dafoe, Andy Lau  2016   
8    Charlie Hunnam, Robert Pattinson, Sienna Mille...  2016   
12   Felicity Jones, Diego Luna, Alan Tudyk, Donnie...  2016   
..                                                 ...   ...   
958  Kevin Costner, Hailee Steinfeld, Connie Nielse...  2014   
968  Anna Hutchison, Andrea Whitburn, Jennifer Koen...  2015   
969  Johnny Depp, Armie Hammer, William Fichtner,To...  2013   
990  Rhona Mitra, Michael Sheen, Bill Nighy, Steven...  2009   
993  Milla Jovovich, Ali Larter, Wentworth Miller,K...  2010   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
4                  123     6.2  393727              325.02       40.0  
5                  103     6.1   56036               45.13       42.0  
8                  141     7.1    7188                8.01       78.0  
12                 133     7.9  323118              532.17       65.0  
..                 ...     ...     ...                 ...        ...  
958                117     6.2   73567               30.69       40.0  
968                 83     3.5    1210                 NaN       37.0  
969                150     6.5  190855               89.29        NaN  
990                 92     6.6  129708               45.80       44.0  
993                 97     5.9  140900               60.13       37.0  

[303 rows x 12 columns]

# 4th dataset

In [18]:
ecom=pd.read_csv('C:/Users/cmscomputers/Downloads/Skype/ecommerce.csv')

In [19]:
ecom.head(10)

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   
5    536365     22752         SET 7 BABUSHKA NESTING BOXES         2   
6    536365     21730    GLASS STAR FROSTED T-LIGHT HOLDER         6   
7    536366     22633               HAND WARMER UNION JACK         6   
8    536366     22632            HAND WARMER RED POLKA DOT         6   
9    536367     84879        ASSORTED COLOUR BIRD ORNAMENT        32   

      InvoiceDate  UnitPrice  CustomerID         Country  
0  12/1/2010 8:26       2.55     17850.0  United Kingdom  
1  12/1/2010 8:26       3.39     17850.0  United Kingdom  
2  12/1/2010 8:26       2.75     17850.0  United Kingdom  
3  12/1/2010 8:26       3.39     17850.0  United Kingdom  
4  12/1/2010 8:26       3.39     17850.0  United Kingdom  
5  12/1/2010 8:26       7.65     17850.0  United Kingdom  
6  12/1/2010 8:26       4.25     17850.0  United Kingdom  
7  12/1/2010 8:28       1.85     17850.0  United Kingdom  
8  12/1/2010 8:28       1.85     17850.0  United Kingdom  
9  12/1/2010 8:34       1.69     13047.0  United Kingdom

In [ ]:
ecom.info()

Find out total puchases made by each customer.

In [20]:
ecom2=(ecom.groupby(by='CustomerID').Quantity.count())
ecom2

CustomerID
12346.0      2
12347.0    182
12348.0     31
12349.0     73
12350.0     17
          ... 
18280.0     10
18281.0      7
18282.0     13
18283.0    756
18287.0     70
Name: Quantity, Length: 4372, dtype: int64

Find out top 20 customers based on number of purchases made

In [21]:
ecom2.sort_values(ascending=False).head(20)

CustomerID
17841.0    7983
14911.0    5903
14096.0    5128
12748.0    4642
14606.0    2782
15311.0    2491
14646.0    2085
13089.0    1857
13263.0    1677
14298.0    1640
15039.0    1508
14156.0    1420
18118.0    1284
14159.0    1212
14796.0    1165
15005.0    1160
16033.0    1152
14056.0    1128
14769.0    1094
17511.0    1076
Name: Quantity, dtype: int64

# 5th dataset

In [22]:
cre=pd.read_csv('C:/Users/cmscomputers/Downloads/Skype/Credit.csv')

In [23]:
cre.head()

Unnamed: 0  ID   Income  Limit  Rating  Cards  Age  Education  Gender  \
0           1   1   14.891   3606     283      2   34         11    Male   
1           2   2  106.025   6645     483      3   82         15  Female   
2           3   3  104.593   7075     514      4   71         11    Male   
3           4   4  148.924   9504     681      3   36         11  Female   
4           5   5   55.882   4897     357      2   68         16    Male   

  Student Married  Ethnicity  Balance  
0      No     Yes  Caucasian      333  
1     Yes     Yes      Asian      903  
2      No      No      Asian      580  
3      No      No      Asian      964  
4      No     Yes  Caucasian      331

In [24]:
cre['Income'].sort_values()

58      10.354
250     10.363
199     10.403
235     10.503
262     10.588
        ...   
275    163.329
261    180.379
355    180.682
323    182.728
28     186.634
Name: Income, Length: 400, dtype: float64

In [26]:
q =[]
for i in cre['Income']:
    if(i <=50):
        q.append('Poor')
    elif(i <=100):
        q.append('Middle')
    elif(i <=150):
        q.append('Upper_Middle')
    else:
        q.append('Rich_affluent')


In [27]:
cre['class']=q

In [29]:
cre.head(15)

Unnamed: 0  ID   Income  Limit  Rating  Cards  Age  Education  Gender  \
0            1   1   14.891   3606     283      2   34         11    Male   
1            2   2  106.025   6645     483      3   82         15  Female   
2            3   3  104.593   7075     514      4   71         11    Male   
3            4   4  148.924   9504     681      3   36         11  Female   
4            5   5   55.882   4897     357      2   68         16    Male   
5            6   6   80.180   8047     569      4   77         10    Male   
6            7   7   20.996   3388     259      2   37         12  Female   
7            8   8   71.408   7114     512      2   87          9    Male   
8            9   9   15.125   3300     266      5   66         13  Female   
9           10  10   71.061   6819     491      3   41         19  Female   
10          11  11   63.095   8117     589      4   30         14    Male   
11          12  12   15.045   1311     138      3   64         16    Male   
12          13  13   80.616   5308     394      1   57          7  Female   
13          14  14   43.682   6922     511      1   49          9    Male   
14          15  15   19.144   3291     269      2   75         13  Female   

   Student Married         Ethnicity  Balance         class  
0       No     Yes         Caucasian      333          Poor  
1      Yes     Yes             Asian      903  Upper_Middle  
2       No      No             Asian      580  Upper_Middle  
3       No      No             Asian      964  Upper_Middle  
4       No     Yes         Caucasian      331        Middle  
5       No      No         Caucasian     1151        Middle  
6       No      No  African American      203          Poor  
7       No      No             Asian      872        Middle  
8       No      No         Caucasian      279          Poor  
9      Yes     Yes  African American     1350        Middle  
10      No     Yes         Caucasian     1407        Middle  
11      No      No         Caucasian        0          Poor  
12      No     Yes             Asian      204        Middle  
13      No     Yes         Caucasian     1081          Poor  
14      No      No  African American      148          Poor

# 6th Dataset

In [3]:
md=pd.read_csv('C:/Users/cmscomputers/Downloads/Skype/mcdonalds.csv')

In [4]:
md.head()

Date        Item  Quantity
0  12-Jun-2020  Aloo tikki         2
1  12-Jun-2020    McVeggie         1
2  12-Jun-2020        Chai         2
3  12-Jun-2020    McVeggie         1
4  12-Jun-2020       Fries         2

 Find out date wise total number of aloo tikkis sold.

In [6]:
m=md[md.Item=='Aloo tikki']

In [12]:
m.groupby(by=['Date'])['Quantity'].sum()

Date
12-Jun-2020     2
13-Jun-2020    18
Name: Quantity, dtype: int64